# vim: se ft=diff :
# HG changeset patch
# User Yonggang Luo <luoyonggang@gmail.com>
# Date 1630660594 -7200
# Parent  3634785345fe6391e073ebf19e3f7ce031a43511
Refinement the nsTreeBodyFrame::GetImage to reduce the calling to mView->GetImageSrc(aRowIndex, aCol, imageSrc);

diff --git a/layout/xul/tree/nsTreeBodyFrame.cpp b/layout/xul/tree/nsTreeBodyFrame.cpp
--- a/layout/xul/tree/nsTreeBodyFrame.cpp
+++ b/layout/xul/tree/nsTreeBodyFrame.cpp
@@ -1849,19 +1849,21 @@ nsITheme* nsTreeBodyFrame::GetTwistyRect
 nsresult nsTreeBodyFrame::GetImage(int32_t aRowIndex, nsTreeColumn* aCol,
                                    bool aUseContext,
                                    ComputedStyle* aComputedStyle,
                                    bool& aAllowImageRegions,
                                    imgIContainer** aResult) {
   *aResult = nullptr;
 
   nsAutoString imageSrc;
-  mView->GetImageSrc(aRowIndex, aCol, imageSrc);
+  if (!aUseContext) {
+    mView->GetImageSrc(aRowIndex, aCol, imageSrc);
+  }
   RefPtr<imgRequestProxy> styleRequest;
-  if (!aUseContext && !imageSrc.IsEmpty()) {
+  if (!imageSrc.IsEmpty()) {
     aAllowImageRegions = false;
   } else {
     // Obtain the URL from the ComputedStyle.
     aAllowImageRegions = true;
     styleRequest =
         aComputedStyle->StyleList()->mListStyleImage.GetImageRequest();
     if (!styleRequest) return NS_OK;
     nsCOMPtr<nsIURI> uri;
